### PR TITLE
usm: http2: use gostreamer

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
Applying https://github.com/DataDog/datadog-agent/pull/18408 changes on http2

### Motivation
Performance improvements 

### Additional Notes
CPU 11% improvement
image

RSS 8% improvement

image

WSS 9% improvement

image

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Review by Yumasi — CHANGES_REQUESTED** [review]
Just a small comment, otherwise looks good

File: pkg/network/encoding/format.go [hunk 1]
```diff
@@ -96,14 +96,7 @@ func FormatConnection(builder *model.ConnectionBuilder, conn network.ConnectionS
...
 	staticTags, dynamicTags = httpEncoder.GetHTTPAggregationsAndTags(conn, builder)
-
-	// TODO: optimize httpEncoder2 to take a writer and use gostreamer
-	http2Stats, _, _ := http2Encoder.GetHTTP2AggregationsAndTags(conn)
-	if len(http2Stats) != 0 {
-		builder.SetHttp2Aggregations(func(b *bytes.Buffer) {
-			b.Write(http2Stats)
-		})
-	}
+	_, _ = http2Encoder.GetHTTP2AggregationsAndTags(conn, builder)
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/encoding/format.go:99]: it's not a side effect
we're not loading into memory the marshaled connections, but instead directly writing it into the writer/builder

[the remark this answers is not in the document]
Comment (Yumasi) [reply pkg/network/encoding/format.go:99]: My main point was that the name does not convey what the function actually does.

[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/encoding/format.go:99]: How about `GenerateHTTP2AggregationsAndTags`?

**Review by Yumasi — APPROVED** [review]

**Pull request merged**